playground/infrastructure/helper.py

```diff
@@ -44,29 +58,194 @@ def find_examples(work_dir: str) -> List[Example]:
+        return tag_object[BEAM_PLAYGROUND]
+
+    return None
+
+
+def _get_supported_categories(categories_path: str) -> List[str]:
+    """
+    Return list of supported categories from categories_path file
+
+    Args:
+        categories_path: path to the file with categories.
+
+    Returns:
+        All supported categories as a list.
+    """
+    with open(categories_path) as supported_categories:
+        yaml_object = yaml.load(supported_categories.read(), Loader=yaml.SafeLoader)
+        return yaml_object[TagFields.CATEGORIES]
+
+
+def _get_example(filepath: str, filename: str, tag: dict) -> Example:
+    """
+    Return an Example by filepath and filename.
+
+    Args:
+         tag: tag of the example.
+         filepath: path of the example's file.
+         filename: name of the example's file.
+
+    Returns:
+        Parsed Example object.
+    """
+    name = _get_name(filename)
+    sdk = _get_sdk(filename)
+    with open(filepath) as parsed_file:
+        content = parsed_file.read()
+
+    return Example(name, "", sdk, filepath, content, "", STATUS_UNSPECIFIED, Tag(**tag))
+
+
+def _validate(tag: dict, supported_categories: List[str]) -> bool:
+    """
+    Validate all tag's fields
+
+    Validate that tag contains all required fields and all fields have required format.
+
+    Args:
+        tag: beam tag to validate.
+        supported_categories: list of supported categories.
+
+    Returns:
+        In case tag is valid, True
+        In case tag is not valid, False
+    """
+    valid = True
+    for field_name in TAG_FIELDS:
+        if tag.get(field_name) is None:
+            logging.error("tag doesn't contain " + field_name + " field: " + tag.__str__())
+            valid = False
+
+    multifile = tag.get(TagFields.MULTIFILE)
+    if (multifile is not None) & (str(multifile).lower() not in ["true", "false"]):
+        logging.error("tag's field multifile is incorrect: " + tag.__str__())
+        valid = False
+
+    categories = tag.get(TagFields.CATEGORIES)
+    if categories is not None:
+        if type(categories) is not list:
+            logging.error("tag's field categories is incorrect: " + tag.__str__())
+            valid = False
+        else:
+            for category in categories:
+                if category not in supported_categories:
+                    logging.error("tag contains unsupported category: " + category)
+                    logging.error("If you are sure that " + category
+                                  + " category should be placed in Beam Playground, you can add it to the `playground/categories.yaml` file")
+                    valid = False
+    return valid
+
+
+def _get_name(filename: str) -> str:
+    """
+    Return name of the example by his filepath.
+
+    Get name of the example by his filename.
+
+    Args:
+        filename: filename of the beam example file.
+
+    Returns:
+        example's name.
+    """
+    return filename.split(os.extsep)[0]
```

Comment:
I would check or try to catch exception if the Len is 0

Reply:
This method is used only to get names of the files which have `.java`/`.py`/`.go` extensions. So these files can't have `len` = 0.